sic_framework/devices/common_naoqi/naoqi_text_to_speech.py

```diff
@@ -118,10 +130,16 @@ def execute(self, message):
         # Set tts parameters for current request
         self.set_params(message)
 
-        if message.animated:
-            self.atts.say(message.text, message.language)
-        else:
-            self.tts.say(message.text)
+        if message == NaoqiTextToSpeechRequest:
+            if message.animated:
+                self.atts.say(message.text, message.language)
+            else:
+                self.tts.say(message.text)
+
+        if message == NaoqiTextToSpeechStopRequest:
+            self.tts.stopAll()
+            self.atts.stopAll()
```

Comment:
I don't think there is a `stopAll()` function for the `ALAnimatedSpeech` module.